src/setfit/trainer.py

```diff
@@ -113,7 +131,27 @@ def _apply_column_mapping(self, dataset: "Dataset", column_mapping: Dict[str, st
         )
         return dataset
 
+    def call_model_init(self):
```

Comment:
Looking ahead to including trials from e.g. `optuna`, should we extend the function signature to accept a `trial` arg?

Reply:
Yes we should but I think it'd be better to do that on hyperparameter search PR once the trail is implemented. What do you think?

Reply:
Sure, that sounds good to me! Let's merge this then :)